tests/test-generators.py

```diff
@@ -26,56 +26,80 @@ def parse_output(result, error_code) -> str:
 
 for g in sorted(glob.glob('generators/*.clvm')):
     print(f"{g}")
-    start_time = time()
-    error_code, result = run_gen(g)
-    run_time = time() - start_time
+    sys.stdout.write("running generator...\r")
+    start_time = perf_counter()
+    error_code, result = run_gen(g, 0, "generators/block-834752.hex")
+    run_time = perf_counter() - start_time
     output = parse_output(result, error_code)
 
-    start_time = time()
-    error_code2, result2 = run_gen(g, MEMPOOL_MODE)
-    run_time2 = time() - start_time
+    sys.stdout.write("running generator (mempool mode) ...\r")
+    sys.stdout.flush()
+    start_time = perf_counter()
+    error_code2, result2 = run_gen(g, MEMPOOL_MODE, "generators/block-834752.hex")
+    run_time2 = perf_counter() - start_time
     output2 = parse_output(result2, error_code2)
 
     with open(g) as f:
         expected = f.read().split('\n', 1)[1]
         if not "STRICT" in expected:
             expected2 = expected
-            if not (result is None and result2 is None or result.cost == result2.cost):
+            if not (result is None or result2 is None or result.cost == result2.cost):
```

Comment:
I'd suggest:

    if result is not None and result2 is not None and result.cost != result2.cost:

It seems a bit easier to understand here.